### PR TITLE
[IMP] shopinvader_sale_coupon: Add remove_coupon route to remove a coupon from code

### PR DESCRIPTION
**Pull request by paradoxxxzero**
This PR adds an utility method to remove a coupon from a cart by specifying the coupon code.
Works for coupon and promotion (check coupon first).

### CODE REVIEW

File: shopinvader_sale_coupon/services/cart.py [hunk 1]
```diff
@@ -34,6 +42,26 @@ def _apply_coupon(self, cart, params):
...
 
+    def _remove_coupon(self, cart, params):
+        code = params["code"]
+        # First we check if this code is a coupon
+        coupon = self.env["coupon.coupon"].search([("code", "=", code)])
+        if coupon:
+            program = coupon.program_id
+        else:
+            # If it's not a coupon, it could be a promo with a code
+            program = self.env["coupon.program"].search([("promo_code", "=", code)])
```
Comment (ivantodorovich) [inline shopinvader_sale_coupon/services/cart.py]: You can probably do this in a single search
```suggestion
        program = self.env["coupon.program"].search(
            [
                "|",
                ("promo_code", "=", code),
                ("coupon_ids.code", "=", code),
            ],
            limit=1,
        )
```

File: shopinvader_sale_coupon/services/cart.py [hunk 2]
```diff
@@ -34,6 +42,26 @@ def _apply_coupon(self, cart, params):
...
+
+        # Next we get the sale order line corresponding to this coupon/promo
+        sol = cart.order_line.filtered(
+            lambda line: line.product_id == program.discount_line_product_id
+        )
```
Comment (ivantodorovich) [inline shopinvader_sale_coupon/services/cart.py]:
```suggestion
        sol = cart.order_line.filtered(
            lambda line: line.is_reward_line and line.product_id == program.discount_line_product_id
        )
```

**Comment by paradoxxxzero** [comment]
Thanks for the review!

**Review by ivantodorovich — APPROVED** [review]

**Comment by paradoxxxzero** [comment]
Is there anything still blocking? (FYI @hparfr )

**Pull request merged**